src/test/java/org/folio/rs/service/RetrievalQueueServiceUnitTest.java

```diff
@@ -0,0 +1,187 @@
+    when(movedEventRequest.getItemStatusName()).thenReturn(ITEM_STATUS_NAME);
+    when(movedEventRequest.getRequesterId()).thenReturn(REQUESTER_ID);
+    when(movedEventRequest.getPickupServicePointId()).thenReturn(PICKUP_LOCATION);
+    when(movedEventRequest.getHoldId()).thenReturn(HOLD_ID);
+    when(movedEventRequest.getRequestNote()).thenReturn(REQUEST_NOTE);
+    when(movedEventRequest.getRequestStatus()).thenReturn(STATUS);
+    when(movedEventMapper.mapDtoToEntity(movedEvent)).thenReturn(movedEventRequest);
+    when(inventoryClient.getItem("barcode==" + ITEM_BARCODE)).thenReturn(items);
+    when(items.getResult()).thenReturn(Collections.singletonList(item));
+    when(item.getTitle()).thenReturn(INSTANCE_TITLE);
+    when(item.getContributorNames()).thenReturn(Collections.singletonList(contributor));
+    when(contributor.getName()).thenReturn(INSTANCE_AUTHOR);
+    when(item.getEffectiveLocation()).thenReturn(effectiveLocation);
+    when(effectiveLocation.getId()).thenReturn(EFFECTIVE_LOCATION_ID);
+    when(item.getEffectiveCallNumberComponents()).thenReturn(callNumberComponents);
+    when(callNumberComponents.getCallNumber()).thenReturn(CALL_NUMBER);
+    when(locationMappingsService.getMappingByFolioLocationId(EFFECTIVE_LOCATION_ID)).thenReturn(locationMapping);
+    when(locationMapping.getConfigurationId()).thenReturn(REMOTE_STORAGE_ID);
+    when(locationClient.getLocation(EFFECTIVE_LOCATION_ID)).thenReturn(folioLocation);
+    when(folioLocation.getCode()).thenReturn(LOCATION_CODE);
+    when(usersClient.query("id==" + REQUESTER_ID)).thenReturn(users);
+    when(users.getResult()).thenReturn(Collections.singletonList(user));
+    when(user.getBarcode()).thenReturn(PATRON_BARCODE);
+    when(user.getUsername()).thenReturn(PATRON_NAME);
+  }
+
+  @Test
+  void shouldSaveMovedEventRequest() {
+    service.processMovedEventRequest(movedEvent);
+
+    verify(retrievalQueueRepository, times(1)).save(captor.capture());
+    RetrievalQueueRecord record = captor.getValue();
+    assertEquals(HOLD_ID, record.getHoldId());
+    assertEquals(ITEM_BARCODE, record.getItemBarcode());
+    assertEquals(CALL_NUMBER, record.getCallNumber());
+    assertEquals(PATRON_BARCODE, record.getPatronBarcode());
+    assertEquals(PATRON_NAME, record.getPatronName());
+    assertEquals(LOCATION_CODE, record.getPickupLocation());
+    assertEquals(STATUS, record.getRequestStatus());
+    assertEquals(REQUEST_NOTE, record.getRequestNote());
+    assertEquals(INSTANCE_TITLE, record.getInstanceTitle());
+    assertEquals(INSTANCE_AUTHOR, record.getInstanceAuthor());
+    assertEquals(REMOTE_STORAGE_ID, record.getRemoteStorageId().toString());
+    assertNotNull(record.getCreatedDateTime());
+  }
+
+  @Test
+  void shouldNotSaveRequestWhenRequestIsNotPaged() {
+    when(movedEventRequest.getItemStatusName()).thenReturn("Hold");
+
+    service.processMovedEventRequest(movedEvent);
+
+    verify(retrievalQueueRepository, never()).save(any());
```

Comment:
any() is general, it is more convenient to use isA()  defining type of object

Reply:
Changed